scripts/o3de/o3de/register.py

```diff
@@ -419,6 +434,11 @@ def register_template_path(json_data: dict,
                            template_path: pathlib.Path,
                            remove: bool = False,
                            engine_path: pathlib.Path = None) -> int:
+    # If a project path or engine path has not been supplied auto detect which manifest to register the input path
+    if not project_path and not engine_path:
+        project_path = utils.find_ancestor_dir_containing_file(pathlib.Path('project.json'), template_path)
+        if not project_path:
+            engine_path = utils.find_ancestor_dir_containing_file('engine.json', template_path)
```

Comment:
Should it be `pathlib.Path('engine.json')` ?

Reply:
It should be, doesn't cause a problem though, because we the `find_ancestor_file` function also calls `pathlib.Path` on the input parameter.
I will update this.